dart/test/monster_test_my_game.example_generated.dart

```diff
@@ -887,6 +1194,371 @@ class Monster {
+  List<int> flex;
+  List<TestT> test5;
+  List<int> vectorOfLongs;
+  List<double> vectorOfDoubles;
+  my_game.InParentNamespaceT parentNamespaceTest;
+  List<ReferrableT> vectorOfReferrables;
+  int singleWeakReference;
+  List<int> vectorOfWeakReferences;
+  List<ReferrableT> vectorOfStrongReferrables;
+  int coOwningReference;
+  List<int> vectorOfCoOwningReferences;
+  int nonOwningReference;
+  List<int> vectorOfNonOwningReferences;
+  AnyUniqueAliasesTypeId anyUniqueType;
+  dynamic anyUnique;
+  AnyAmbiguousAliasesTypeId anyAmbiguousType;
+  dynamic anyAmbiguous;
+  List<Color> vectorOfEnums;
+  Race signedEnum;
+  List<int> testrequirednestedflatbuffer;
+  List<StatT> scalarKeySortedTables;
+
+  MonsterT({
+      this.pos,
+      this.mana,
+      this.hp,
+      this.name,
+      this.inventory,
+      this.color,
+      this.testType,
+      this.test,
+      this.test4,
+      this.testarrayofstring,
+      this.testarrayoftables,
+      this.enemy,
+      this.testnestedflatbuffer,
+      this.testempty,
+      this.testbool,
+      this.testhashs32Fnv1,
+      this.testhashu32Fnv1,
+      this.testhashs64Fnv1,
+      this.testhashu64Fnv1,
+      this.testhashs32Fnv1a,
+      this.testhashu32Fnv1a,
+      this.testhashs64Fnv1a,
+      this.testhashu64Fnv1a,
+      this.testarrayofbools,
+      this.testf,
+      this.testf2,
+      this.testf3,
+      this.testarrayofstring2,
+      this.testarrayofsortedstruct,
+      this.flex,
+      this.test5,
+      this.vectorOfLongs,
+      this.vectorOfDoubles,
+      this.parentNamespaceTest,
+      this.vectorOfReferrables,
+      this.singleWeakReference,
+      this.vectorOfWeakReferences,
+      this.vectorOfStrongReferrables,
+      this.coOwningReference,
+      this.vectorOfCoOwningReferences,
+      this.nonOwningReference,
+      this.vectorOfNonOwningReferences,
+      this.anyUniqueType,
+      this.anyUnique,
+      this.anyAmbiguousType,
+      this.anyAmbiguous,
+      this.vectorOfEnums,
+      this.signedEnum,
+      this.testrequirednestedflatbuffer,
+      this.scalarKeySortedTables});
+
+  int pack(fb.Builder fbBuilder) => _pack(fbBuilder, HashMap<Object, int>());
+
+  /// Write the object to the [Builder] only once and reuse the same offset on
+  /// subsequent references of the same object.
+  ///
+  /// Note that this method assumes you call it using the same [Builder]
+  /// instance every time. The returned offset is only good for the [Builder]
+  /// used in the first call to this method.
+  int packOnce(fb.Builder fbBuilder, Map<Object, int> existingOffsets) {
+    assert(existingOffsets != null);
+    return existingOffsets[this] ??= _pack(fbBuilder, existingOffsets);
+  }
+
+  int _pack(fb.Builder fbBuilder, Map<Object, int> existingOffsets) {
+    assert(existingOffsets != null);
+    assert(fbBuilder != null);
+    final int nameOffset = fbBuilder.writeString(name);
+    final int inventoryOffset = inventory?.isNotEmpty == true
```

Comment:
isn't `== true` superfluous? as well as `!= null` elsewhere etc?

Reply:
the `?` means it could be null.

Reply:
to expand: without null-safety in place, if `inventory` is null, the whole expression is null. If ` == true` was left out, it wouldn't be a valid code: "The value of the '?.' operator can be 'null', which isn't appropriate in a condition."